test-runner-poc/lib/runner.js

```diff
@@ -1,73 +1,148 @@
-"use strict";
+'use strict';
 
-const path = require("path");
-const fs = require("fs");
-const utils = require("mocha").utils;
+const path = require('path');
+const fs = require('fs');
+const utils = require('mocha').utils;
+const MongoClient = require('mongodb').MongoClient;
+const parseConnectionString = require('../../lib/core/uri_parser');
 
+const testPath = path.join(path.join(process.cwd(), '../'), 'test');
+const configPath = path.join(testPath, 'config.js');
+const envPath = path.join(testPath, 'environments.js');
+const environments = require(envPath);
+const TestConfiguration = require(configPath);
+
+let mongoClient;
 let filters = [];
-let files = [];
+let initializedFilters = 0;
 
 function addFilter(filter) {
-	if (typeof filter !== "function" && typeof filter !== "object") {
+	if (typeof filter !== 'function' && typeof filter !== 'object') {
 		throw new Error(
-			"Type of filter must either be a function or an object"
+			'Type of filter must either be a function or an object'
 		);
 	}
 	if (
-		typeof filter === "object" &&
-		(!filter.filter || typeof filter.filter !== "function")
+		typeof filter === 'object' &&
+		(!filter.filter || typeof filter.filter !== 'function')
 	) {
-		throw new Error("Object filters must have a function named filter");
+		throw new Error('Object filters must have a function named filter');
 	}
 
-	if (typeof filter === "function") {
+	if (typeof filter === 'function') {
 		filters.push({ filter: filter });
 	} else {
 		filters.push(filter);
 	}
 }
 
-before(function() {
+function findMongo(packagePath) {
```

Comment:
This function originally existed b/c you could potentially have `mongodb` or `mongodb-core`. There is only one `mongo` to look for right now, so this should not be necessary.

Reply:
removed the findMongo function